MoreLinq/CopyTo.cs

```diff
@@ -0,0 +1,172 @@
+                }
+
+                return i - index;
+            }
+        }
+
+        /// <summary>
+        /// Copies the contents of a sequence into a provided list.
+        /// </summary>
+        /// <typeparam name="TSource">
+        /// The type of elements of <paramref name="source"/></typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="list">The list that is the destination of the elements copied from <paramref
+        /// name="source"/>.</param>
+        /// <returns>The number of elements actually copied.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="list"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        /// <para>
+        /// All data from <paramref name="source"/> will be copied to <paramref name="list"/>, starting at position 0.
+        /// </para>
+        /// <para>
+        /// This operator executes immediately.
+        /// </para>
+        /// </remarks>
+        public static int CopyTo<TSource>(this IEnumerable<TSource> source, IList<TSource> list)
+        {
+            return source.CopyTo(list, 0);
+        }
+
+        /// <summary>
+        /// Copies the contents of a sequence into a provided list.
+        /// </summary>
+        /// <typeparam name="TSource">
+        /// The type of elements of <paramref name="source"/></typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="list">The list that is the destination of the elements copied from <paramref
+        /// name="source"/>.</param>
+        /// <param name="index">The position in <paramref name="list"/> at which to start copying data</param>
+        /// <returns>The number of elements actually copied.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="list"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        /// <para>
+        /// All data from <paramref name="source"/> will be copied to <paramref name="list"/>, starting at position
+        /// <paramref name="index"/>.
+        /// </para>
+        /// <para>
+        /// This operator executes immediately.
+        /// </para>
+        /// </remarks>
+        public static int CopyTo<TSource>(this IEnumerable<TSource> source, IList<TSource> list, int index)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (list == null) throw new ArgumentNullException(nameof(list));
+            if (index < 0) throw new ArgumentOutOfRangeException(nameof(index));
+
+            if (list is TSource[] array)
+            {
+                return CopyTo(source, array, index);
+            }
+            else
```

Comment:
💡You don't need the else statement here, because there is no more instruction after it.